allensdk/internal/morphology/morphology.py

```diff
@@ -808,7 +808,7 @@ def apply_affine(self, aff, scale=None):
         #
         if scale is None:
             # calculate the determinant
-            determinant = np.linalg.det(np.reshape(aff, (3, 3)))
+            determinant = np.linalg.det(np.reshape(aff[0:9], (3, 3)))
```

Comment:
Since scaling is determined by the affine transform, it makes sense to add scaling calculation to the affine transform class. Also, it would be cleaner, if the inline comment starting on line 799 describing the reasons for calculating determinant was instead included into the docstring for the function computing scaling.

Reply:
I did both of those things in the neuron_morphology repo, but I didn't want to mess with this code too much

Reply:
Totally agree. I have missed on the fact that this is not a neuron_morphology.